lib/Notes/utils/tests/getHTMLSubstring-test.js

```diff
@@ -0,0 +1,56 @@
+import {
+  describe,
+  it,
+} from '@bigtest/mocha';
+import { expect } from 'chai';
+
+import { getHTMLSubstring } from '..';
+import { DETAILS_CUTOFF_LENGTH } from '../../constants';
+
+const getString = (length, character) => new Array(length).fill(character).join('');
+
+const createDOMElement = domObject => {
```

Comment:
I think we shouldn't write any logic inside tests, because this may also be buggy and could cause unstable tests.
If I understood correctly what you're trying to do here - is generate a html string. If that's so then I'd prefer to do it by using string templates
For example:
```
const inputHTMLString = `
  <ul>
    <li>${getString(100, 'a')}</li>
    <li>${getString(100, 'b')}</li>
    <li>${getString(50, 'c')}${getString(50, 'd')}</li>
  </ul>
`
```
This accomplishes the same thing, but you don't need to understand `getStringFromElement` function to get what the test does and there wouldn't be a situation where you have to debug test code

Reply:
Yes, but if we write as you indicated, then there is a space between each line, it is also considered as a character, so mocked lines need to be written in one line, which is why we will have a lint error(string-length). We can more simplest string, but it wouldn't cover all conditions. Additional functionality has been added in case we have to transfer dates of different sizes, configurations and can be easily scaled in the future.

Reply:
@Vladyslav-Velytskyi, @BogdanDenis, I agree with Denis.  Too much code to grasp before moving on to the tests themselves.
Vlad, if you are concerned about huge html string, I'd suggest to move them to separate files. Not an elegant solution, but I think this is a reasonable price for having straightforward tests.

Reply:
Done
@BogdanDenis , @MaksymDryha please review it